### PR TITLE
[CBRD-24479] The maximum number of objects in lockdb is meaningless

### PR DESCRIPTION
**Pull request by shparkcubrid**
http://jira.cubrid.org/browse/CBRD-24479

add 'Current number of objects which are allocated'
add 'Current size of objects which are allocated '
add -c (contention only) mode

### CODE REVIEW

File: src/executables/util_front.c [hunk 1]
```diff
@@ -137,6 +137,7 @@ static ARG_MAP_TABLE ua_Space_map[] = {
 
 static ARG_MAP_TABLE ua_Lock_map[] = {
   {"-o", "--" LOCK_OUTPUT_FILE_L},
+  {"-a", "--" LOCK_DISPLAY_CONTENTION_L},
```
Comment (shparkcubrid) [inline src/executables/util_front.c]:
```suggestion
  {"-c", "--" LOCK_DISPLAY_CONTENTION_L},
```

**Review by youngjinj — APPROVED** [review]

File: src/transaction/lock_manager.c [hunk 2]
```diff
@@ -8518,18 +8518,37 @@ xlock_dump (THREAD_ENTRY * thread_p, FILE * outfp)
...
   num_locked = (int) lk_Gl.m_obj_hash_table.get_element_count ();
+  num_resource_alloc = (int) lk_Gl.m_obj_hash_table.get_alloc_element_count ();
+  num_entry_alloc = (int) lk_Gl.obj_free_entry_list.alloc_cnt;
+  size_alloc = (num_entry_alloc * sizeof (LK_ENTRY)) + (num_resource_alloc * sizeof (LK_RES));
 
   /* dump object lock table */
   fprintf (outfp, "Object Lock Table:\n");
   fprintf (outfp, "\tCurrent number of objects which are locked    = %d\n", num_locked);
-  fprintf (outfp, "\tMaximum number of objects which can be locked = %d\n\n", lk_Gl.max_obj_locks);
+  fprintf (outfp, "\tCurrent number of objects which are allocated = %d\n", num_resource_alloc);
```
Comment (beyondykk9) [inline src/transaction/lock_manager.c:8529]:
```suggestion
  fprintf (outfp, "\tCurrent number of objects which are allocated = %dBytes\n", num_resource_alloc);
```

**Review by beyondykk9 — APPROVED** [review]

File: src/transaction/lock_manager.c (hunk 2 above)
Comment (shparkcubrid) [reply src/transaction/lock_manager.c:8529]: This is the number of locks allocated in memory.

**Review by xmilex-git — APPROVED** [review]

File: src/communication/network_interface_cl.c [hunk 3]
```diff
@@ -3881,22 +3881,31 @@ acl_dump (FILE * outfp)
...
   if (outfp == NULL)
     {
       outfp = stdout;
     }
 
-  req_error = net_client_request_recv_stream (NET_SERVER_LK_DUMP, NULL, 0, NULL, 0, NULL, 0, outfp);
+  request = OR_ALIGNED_BUF_START (a_request);
+
+  ptr = request;
+  ptr = or_pack_int (ptr, is_contention);
```
Comment (hornetmj) [inline src/communication/network_interface_cl.c]:
```suggestion
  (void) or_pack_int (request, is_contention);
```

File: src/transaction/lock_manager.c [hunk 4]
```diff
@@ -8518,18 +8518,37 @@ xlock_dump (THREAD_ENTRY * thread_p, FILE * outfp)
...
+      fprintf (outfp, "\tCurrent size of objects which are allocated = %d\n\n", size_alloc);
+    }
+  else if (size_alloc >= 1024 && size_alloc < 1048576)
+    {
+      fprintf (outfp, "\tCurrent size of objects which are allocated = %dK\n\n", size_alloc / 1024);
+    }
+  else
+    {
+      fprintf (outfp, "\tCurrent size of objects which are allocated = %dM\n\n", size_alloc / 1024 / 1024);
+    }
```
Comment (hornetmj) [inline src/transaction/lock_manager.c]:
```suggestion
  if (size_alloc < ONE_K)
    {
      fprintf (outfp, "\tCurrent size of objects which are allocated = %d\n\n", size_alloc);
    }
  else if (size_alloc >= ONE_K && size_alloc < ONE_M)
    {
      fprintf (outfp, "\tCurrent size of objects which are allocated = %dK\n\n", size_alloc / ONE_K);
    }
  else
    {
      fprintf (outfp, "\tCurrent size of objects which are allocated = %dM\n\n", size_alloc / ONE_M);
    }
```

File: src/transaction/lock_manager.c [hunk 5]
```diff
@@ -8456,7 +8456,7 @@ xlock_dump (THREAD_ENTRY * thread_p, FILE * outfp)
   int old_wait_msecs = 0;	/* Old transaction lock wait */
   int tran_index;
   LK_RES *res_ptr;
-  int num_locked;
+  int num_locked, num_entry_alloc, num_resource_alloc, size_alloc;
```
Comment (hornetmj) [inline src/transaction/lock_manager.c]: How about using `UINT64 size_alloc`?

**Review by hornetmj — COMMENTED** [review]
Still reviewing.

File: src/transaction/lock_manager.c (hunk 5 above)
Comment (shparkcubrid) [reply src/transaction/lock_manager.c]: I fixed it.

File: src/transaction/lock_manager.c [hunk 6]
```diff
@@ -8518,18 +8519,37 @@ xlock_dump (THREAD_ENTRY * thread_p, FILE * outfp)
...
+    {
+      fprintf (outfp, "\tCurrent size of objects which are allocated = %d\n\n", size_alloc);
+    }
+  else if (size_alloc >= ONE_K && size_alloc < ONE_M)
+    {
+      fprintf (outfp, "\tCurrent size of objects which are allocated = %dK\n\n", size_alloc / ONE_K);
+    }
+  else
+    {
+      fprintf (outfp, "\tCurrent size of objects which are allocated = %dM\n\n", size_alloc / ONE_M);
```
Comment (hornetmj) [inline src/transaction/lock_manager.c]:
```suggestion
      fprintf (outfp, "\tCurrent size of objects which are allocated = %llu\n\n", size_alloc);
    }
  else if (size_alloc >= ONE_K && size_alloc < ONE_M)
    {
      fprintf (outfp, "\tCurrent size of objects which are allocated = %lluK\n\n", size_alloc / ONE_K);
    }
  else
    {
      fprintf (outfp, "\tCurrent size of objects which are allocated = %lluM\n\n", size_alloc / ONE_M);
```

**Review by hornetmj — APPROVED** [review]
LGTM.

**Pull request merged**